### PR TITLE
Removing postcode lookup api key from repo.

### PR DESCRIPTION
**Pull request by zaparka**
Loading it from ENV as it should be done in first place.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -71,12 +71,15 @@ Feature tests should be run on localhost against the instance running in the con
...
 
 ### Dependencies
 * Virtual Box  - install from https://www.virtualbox.org/wiki/Downloads
 * Vagrant - install from http://www.vagrantup.com/downloads.html
 
-
+# Postcode lookup
+* We are using Ideal postcodes.
+* API key is set in ENV
+* Account info is here https://tools.hmcts.net/confluence/display/RST/AP+URLs+and+Analytics
```
Comment (leoapost) [inline README.md]: I don't think this info is that useful to others to be in the README and the URL could change at any time. How about to remove the line?

File: config/initializers/ideal_postcode.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+ENV['POSTCODE_LOOKUP_API_URL'] = 'https://api.ideal-postcodes.co.uk/v1/postcodes/'
```
Comment (leoapost) [inline config/initializers/ideal_postcode.rb]: How about to use the `.env` file instead? This way we could define sandboxes for dev envs etc

File: README.md (hunk 1 above)
Comment (zaparka) [reply README.md]: @leoapost I disagree. I think we should keep it here so we have at least some reference. We are using Confluence as documentation so the link should not change anytime soon.

File: config/initializers/ideal_postcode.rb (hunk 2 above)
Comment (zaparka) [reply config/initializers/ideal_postcode.rb]: Done

File: README.md (hunk 1 above)
Comment (leoapost) [reply README.md]: Sorry for the confusion, maybe I am missing something here.. My thought was that where to find info about our private Ideal account is an internal matter and should be enough for us to just add it in our private Wiki where we keep similar info for the project.

I don't see how having it in the README can be useful to anyone outside our team. If someone wants to use this code, they will have to create their own account.

I hope it makes sense :)

@garytaylor what's your thoughts?

Comment (garytaylor) [reply README.md]: I agree with both of you lol.

It is important that anyone coming on board in the future doesn't end up having to ask lots of people just to get setup on a project..  Sure, it should say that you need a postcode lookup account, but it should also say if you are a member of this team, you can find the team's account details here.

File: app/lib/postcode_lookup_proxy.rb [hunk 3]
```diff
@@ -17,10 +17,9 @@ def initialize(postcode, valid_countries, use_live_data = false)
     @postcode                 = UKPostcode.parse(postcode)
     @valid                    = @postcode.valid?
     @result_set               = nil
-    config                    = YAML.load_file("#{Rails.root}/config/ideal_postcodes.yml")
-    @url                      = config['url']
-    @api_key                  = config['api_key']
-    @timeout                  = config['timeout']
+    @url                      = ENV['POSTCODE_LOOKUP_API_URL']
+    @api_key                  = ENV['POSTCODE_LOOKUP_API_KEY']
+    @timeout                  = ENV['POSTCODE_LOOKUP_TIMEOUT'].to_f
```
Comment (garytaylor) [inline app/lib/postcode_lookup_proxy.rb]: Will this error if the ENV var is not set ?

File: config/initializers/ideal_postcode.rb [hunk 4]
```diff
@@ -0,0 +1 @@
+ENV['POSTCODE_LOOKUP_TIMEOUT'] = '3.0'
```
Comment (garytaylor) [inline config/initializers/ideal_postcode.rb]: We shouldn't be modifying ENV - only read from it surely ?  You are effectively modifying the input to the process which seems wrong

File: README.md (hunk 1 above)
Comment (zaparka) [reply README.md]: @leoapost I removed the direct link to the page and left just link to the confluence.

File: config/initializers/ideal_postcode.rb (hunk 4 above)
Comment (leoapost) [reply config/initializers/ideal_postcode.rb]: I agree. This doesn't allow to dynamically set the variable, so maybe move it also in the .env file? If we don't need this to be dynamic then I guess it doesn't have to be an ENV variable.

**Comment by zaparka** [comment]
@leoapost @garytaylor I removed the timeout from env and updated the readme.

**Pull request merged**